common/src/main/java/whocraft/tardis_refined/registry/ARSStructurePieceRegistry.java

```diff
@@ -0,0 +1,108 @@
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r356")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r39")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r408")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r412")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r416")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r479")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r5")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r508")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r527")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r536")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r611")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r628")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r642")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r649")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r658")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r677")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r684")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r687")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r719")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r73")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r794")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r814")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r851")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r904")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r917")));
+        registerRoomPiece(new ARSStructurePiece(createRoomResourceLocation("gs_r994")));
+    }
+
+
+    /**
+     * Register a new corridor to be featured in corridor generation.
+     * @param corridorPiece ARSStructurePiece to be placed.
+     */
+
+    private static void registerCorridorPiece(ARSStructurePiece corridorPiece) {
+        CORRIDORS.add(corridorPiece);
+    }
+
+    /**
+     * Register a new room to be featured in corridor generation.
+     * @param roomPiece ARSStructurePiece to be placed.
+     */
+
+    private static void registerRoomPiece(ARSStructurePiece roomPiece) {
+        ROOMS.add(roomPiece);
+    }
+
+    private static ResourceLocation createCorridorResourceLocation(String id) {
```

Comment:
If we would like other mods to add their own room types and corridors, we can consider adding a version of these Resource locations creation methods that aren't hardcoded to the TARDIS Refined namespace.

Reply:
I have made the registry functions public, however I have kept the create resource locations private and linked to our mod id. This is a small helper function for us, but we don't have to restrict other devs to their own file structure.